apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -516,12 +521,18 @@ export default class InstallManager {
         return;
       }
     } else {
-      console.log(`Deleting the NPM cache folder`);
-      // This is faster and more thorough than "npm cache clean"
-      this._asyncRecycler.moveFolder(this._rushConfiguration.npmCacheFolder);
+      if (this._rushConfiguration.packageManager === 'npm') {
+        console.log(`Deleting the packages-cache folder`);
+        // This is faster and more thorough than "npm cache clean"
+        this._asyncRecycler.moveFolder(this._rushConfiguration.npmCacheFolder);
+
+        console.log(`Deleting the "packages-tmp" folder`);
```

Comment:
>"packages-tmp" [](start = 34, length = 14)

Don't rename this.  Don't try to share the same folder name for the NPM and PNPM caches.

Reply:
We don't want:

1. After I migrate my repo to PNPM, everybody's "rush install" is failing due to various corruption.
2. After I migrate back to NPM because I had trouble with PNPM, even more corruption!

---
In reply to: [151785792](https://github.com/Microsoft/web-build-tools/pull/427#discussion_r151785792) [](ancestors = 151785792)